Rubberduck.Core/UI/UnitTesting/TestExplorerControl.xaml.cs

```diff
@@ -68,5 +69,31 @@ protected virtual void Dispose(bool disposing)
             DataContextChanged -= TestExplorerControl_DataContextChanged;
             _isDisposed = true;
         }
+        private GroupingGrid _newGroupingGrid;
+        private void ResultsGroupingGrid_IsVisibleChanged(object sender, System.Windows.DependencyPropertyChangedEventArgs e)
+        {
+            if (sender is GroupingGrid)
+            {
+                GroupingGrid groupingGrid = (GroupingGrid)sender;
+                // actual height checks that form is visible at this point
+                // don't set up the "new" visible element until we see an object with an actual height
+                if (!((groupingGrid.ActualHeight == 0) && (_newGroupingGrid is null))) // check form elements are actually set up (i.e. not initialising form)
+                {
+                    // this code relies on the new element being made visible before the old is hidden, so we save the new element 
+                    // and then copy column widths from the old now hidden element on the subsequent invocation
+                    if (groupingGrid.IsVisible)
+                        _newGroupingGrid = groupingGrid;
+                    else
+                    {
+                        // copy column widths from previously visible grouping grid to ensure widths are consistent
+                        for (int columnIndex = 0; columnIndex < groupingGrid.Columns.Count; columnIndex++)
+                        {
+                            // copy column widths from current now hidden columns to the new visible columns
+                            _newGroupingGrid.Columns[columnIndex].Width = groupingGrid.Columns[columnIndex].Width;
```

Comment:
This line throws an NRE on shutdown.

Reply:
I've not seen an examples of this causing problems on shutdown although admittedly I've only been testing using Excel as a host and my tests have been rather simplistic. Can you suggest how to reproduce?

Reply:
I think to see this, you need to attach a debugger and try to shutdown Excel. The exception is likely being caught and handled (but in suboptimal manner).

Reply:
Ignore that - just found how to recreate.